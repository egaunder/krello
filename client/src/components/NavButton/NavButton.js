import React from 'react'
import PropTypes from 'prop-types'

import './NavButton.css'

const NavButton = props => (
  <div className="nav-button-wrapper">
    {props.render()}
  </div>
)

NavButton.propTypes = { render: PropTypes.func.isRequired }

export default NavButton
