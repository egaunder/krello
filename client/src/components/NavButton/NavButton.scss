.nav-button-wrapper  {
  button {
    box-shadow: 0 2px 0 rgba(0,0,0,0.3);
    border-radius: .4em;
    border: none;
    height: 3em;
    font-size: 1em;
    margin-left: 5px;
    margin-right: 20px;
    width: 5em;
  }
}