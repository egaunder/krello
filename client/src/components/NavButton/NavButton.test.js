import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import NavButton from './NavButton'

describe('NavButton', () => {
  test('should render correctly', () => {
    const render = jest.fn();
    const output = shallow(<NavButton render={render} />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  test('should render button when passed as props', () => {
    const render = () => <button>Test Button</button>
    const wrapper = mount(<NavButton render={render} />)
    const button = wrapper.find('button')
    expect(button.length).toBe(1)
  })
})