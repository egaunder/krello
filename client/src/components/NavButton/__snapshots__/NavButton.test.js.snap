// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton should render correctly 1`] = `
<div
  className="nav-button-wrapper"
/>
`;
